src/server/ocpp/services/OCPPService.ts

```diff
@@ -150,7 +151,23 @@ export default class OCPPService {
           (chargingStation.chargePointSerialNumber && bootNotification.chargePointSerialNumber &&
             chargingStation.chargePointSerialNumber !== bootNotification.chargePointSerialNumber)) {
           // Not the same Charging Station!
-          // FIXME: valid charging stations in DB with modified parameters cannot be registered again without direct access to the DB
+          // FIXME: valid charging stations in DB with modified parameters cannot be registered at boot notification again without direct
+          //        access to the DB, standard charging station replacement is then not possible. The registration status returned should
+          //        be 'Pending' and a way to manually accept or refuse such stations should be offered.
+          // Do not flag charging stations online with registered boot notification as 'Rejected'
+          if (moment().subtract(Configuration.getChargingStationConfig().maxLastSeenIntervalSecs, 'seconds').isSameOrBefore(chargingStation.lastSeen) && chargingStation.registrationStatus === RegistrationStatus.ACCEPTED) {
+            await Logging.logWarning({
+              tenantID: headers.tenantID,
+              source: chargingStation.id,
+              action: ServerAction.BOOT_NOTIFICATION,
+              module: MODULE_NAME, method: 'handleBootNotification',
+              message: 'Trying to connect a charging station matching an online charging station with identical chargeBoxID, registered boot notification and different attributes',
+              detailedMessages: { headers, bootNotification }
+            });
+          // Set charging station boot notification registration status to 'Rejected'
+          } else {
+            await ChargingStationStorage.saveChargingRegistrationStatus(headers.tenantID, chargingStation.id, { registrationStatus: RegistrationStatus.REJECTED });
```

Comment:
I would not also take care of this scenario.
If there is a network outage or some works like in Caen and I connect a charging station with the same ID you will save the status and we should not.
The lastSeen should not be used to assume that this is a charger's replacement.
And the waring message is quite the same as the exception below.
I will not save anything again in this method!

Reply:
The goal is to save the boot notification status in that case to ensure charging station with a rejected status are not seen online. 
The way it's now done after your fruitful code review in that PR handle all use case when charging stations have the same ID properly: 
+ recorded charging station boot notification status is never overridden if the charging station is online and tenant administrator have a warning log on the same ID usage;
+ recorded charging station boot notification status only is overridden if the charging station is offline to ensure station with the same ID but different attributes will not be seen as online ans usable. If the recorded charging station comes back online, the boot registration status is then saved as 'Accepted' and the code behaviour goes back to the previous point. 

If necessary, I can explain next week in the real world the user experience 'bug' that PR is meant to fix. 
	 
And the one to one remplacement scenario handling asked by all of your charging station customers is not handled in that PR.

Reply:
> * recorded charging station boot notification status is only overridden if the charging station is offline to ensure station with the same ID but different attributes will not be seen as online and usable. If the recorded charging station comes back online, the boot registration status in the DB record is then saved as 'Accepted' and the code behaviour goes back to the previous point.

Wrong here: as a general assumption, you cannot change a property which does not belong to the entity.
S/N or model is different -> there is nothing to update here.
Moreover, if there is a network cut, the charging station will reconnect without sending a BootNotification.

> And the one to one remplacement scenario handling asked by all of your charging station integrator customers is not handled in that PR.

This can be done by renaming the old charging station to free the ID with all its history and we didn't do it yet because we simply lacked time. That doesn't mean we will not do it. We have our prios.

Reply:
> > * recorded charging station boot notification status is only overridden if the charging station is offline to ensure station with the same ID but different attributes will not be seen as online and usable. If the recorded charging station comes back online, the boot registration status in the DB record is then saved as 'Accepted' and the code behaviour goes back to the previous point.
> 
> Wrong here: as a general assumption, you cannot change a property which does not belong to the entity.
> S/N or model is different -> there is nothing to update here.
> Moreover, if there is a network cut, the charging station will reconnect without sending a BootNotification.

The issue rely on the data model design for the charging stations collection, the record unicity is only enforced on the chargeboxid. The unicity must be on the tuple (id, vendor, model, serial) to match the intended behaviour and allow record creation in DB and boot notification status proper tracking of charging stations with the same id. 

I've removed for now the saving of the boot registration status in case of failure in that PR, let the boot notification status recording even if not yet useful and will create an issue and implement the clean solution in subsequents PRs that will make use of the boot notification status properly.